File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        word1_length = len(word1)
        word2_length = len(word2)
        index1 = 0
        index2 = 0
        output = ""
        while index1 < word1_length or index2 < word2_length:
            if index1 < word1_length:
                output += word1[index1]
                index1 += 1
            if index2 < word2_length:
                output += word2[index2]
                index2 += 1
        return output
            
        

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.cpp
class Solution {
public:
    string mergeAlternately(string word1, string word2) {
        int m = word1.size();
        int n = word2.size();
        string output = "";
        int i = 0, j = 0;
        
        while(i<m || j<n){
            if(i<m){
                output.push_back(word1[i++]);
            }
            if(j<n){
                output.push_back(word2[j++]);
            }
        }
        return output;
    }
};